app/src/main/java/org/mozilla/fenix/home/sessioncontrol/viewholders/TabViewHolder.kt

```diff
@@ -76,13 +109,49 @@ class TabViewHolder(
         this.tab = tab
         updateTabUI(tab)
         item_tab.transitionName = "$TAB_ITEM_TRANSITION_NAME${tab.sessionId}"
+
+        containerView?.context!!.components.core.sessionManager.findSessionById(tab.sessionId)?.register(mediaObserver)
```

Comment:
In general, `!!` should be avoided like the plague. If an instance is non-null then we know we can safely use it all the time, otherwise use safe calls on it, if you really need it non-null and have reasonable certainty it's safe, try to find a `require` version that throws exceptions that can be better traced out.

Here, the context can be null if the `containerView` is destroyed far sooner than we can get the context from it.

Think of an Android context as 'the lifetime of an android thing', and you can't control when the lifetime of that thing will be destroyed.